subiquity/server/controllers/filesystem.py

```diff
@@ -150,27 +150,31 @@ def setup_disk_for_guided(self, target, mode):
         if DeviceAction.TOGGLE_BOOT in DeviceAction.supported(disk):
             self.add_boot_disk(disk)
         if gap is None:
-            return disk, gaps.largest_gap(disk)
+            return gaps.largest_gap(disk)
         else:
             # find what's left of the gap after adding boot
             gap = gaps.within(disk, gap)
             if gap is None:
                 raise Exception('failed to locate gap after adding boot')
-            return disk, gap
+            return gap
+
+    def guided_partition_creator(self, gap, spec):
+        flag = None
+        if gap.in_extended:
+            flag = 'logical'
+        return self.create_partition(
```

Comment:
Aiee duelling abstractions I guess -- but maybe create_partition should handle (or at least check?) this logic?

Reply:
That would let you delete this method, which has a funny name. This little corner doesn't feel completely thought out?

Reply:
OK, I will move this to `create_partition()`.  I actually started with that.
The lower-level `add_partition()` is a bit too clever for my liking (rounds up sizes!), so maybe we start to make `create_partition()` be the smart one.